packages/components/app/styles/components/code-block/index.scss

```diff
@@ -0,0 +1,209 @@
+.hds-code-block__header {
+  display: flex;
+  flex-flow: column;
+  gap: 4px;
+  padding: 8px $code-block-code-padding;
+  background-color: var(--code-block-color-surface-faint);
+  border-bottom: 1px solid var(--code-block-color-border-strong);
+}
+
+.hds-code-block__description { color: var(--code-block-color-foreground-faint); }
+
+// Code
+.code-block__code {
+  display: block;
+  margin: 0;
+  padding: $code-block-code-padding;
+  overflow: auto;
+}
+
+// CopyButton
+.hds-code-block__copy-button {
+  // Overriding default colors
+  color: var(--code-block-color-foreground-primary);
+  background-color: var(--code-block-color-surface-faint);
+  border: 1px solid var(--code-block-color-border-strong);
+
+  &:hover,
+  &.mock-hover {
+    background-color: var(--code-block-color-surface-primary);
+    border-color: var(--code-block-color-border-strong);
+  }
+
+  &:active,
+  &.mock-active {
+    background-color: #2b303c; // TODO: can't find color defined in Figma
+  }
+
+  &:focus,
+  &.mock-focus,
+  &:focus-visible {
+    background-color: var(--code-block-color-surface-faint);
+  }
+
+  &.hds-copy-button--status-success {
+    .flight-icon {
+      color: var(--code-block-color-foreground-success);
+    }
+  }
+
+  &.hds-copy-button--status-error {
+    .flight-icon {
+      color: var(--code-block-color-foreground-critical);
+    }
+  }
+
+  .flight-icon { color: var(--code-block-color-foreground-primary); }
+}
+
+// VARIANTS
+
+// hasCopyButton
+.hds-code-block--has-copy-button {
+  .hds-code-block__copy-button {
+    position: absolute;
+    top: 8px;
+    right: 8px;
+  }
+}
+
+// LineWrapping
+.hds-code-block--has-line-wrapping {
+  pre,
+  code { white-space: pre-wrap; }
+}
+
+// Linenumbers
+// TODO: Add variant class name to scope (instead of base class name)?
+.hds-code-block {
+  // LineNumbers plugin styles ---------------
+  &.line-numbers {
+    counter-reset: linenumber;
+
+    .code-block__code {
+      position: relative;
+      // reserve space for line numbers
+      padding-left: calc(#{$code-block-line-numbers-width} + #{$code-block-code-padding});
+    }
+
+    .line-numbers-rows {
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      width: $code-block-line-numbers-width;
+      padding: $code-block-code-padding 0;
+      border: 1px solid var(--code-block-color-border-strong);
+      user-select: none;
+      pointer-events: none;
+
+      > span {
+        display: block;
+        counter-increment: linenumber;
+
+        &::before {
+          display: block;
+          padding-right: $code-block-code-padding;
+          text-align: right;
+          content: counter(linenumber);
+        }
+      }
+    }
+  }
+
+  .line-numbers-rows .line-numbers-rows > span
+  // LineHighlight plugin styles (works with lineNumbers) ------------
+  pre[data-line] {
+    position: relative;
+    padding: 1em 0 1em 3em;
+  }
+
+  // Highlighted Lines
+  .line-highlight {
```

Comment:
I'm not really sure what the problem is here, but the line highlight _appears_ to be a little bit lower than the center of the line (I think around 2px from testing). 
<img width="517" alt="Screenshot 2023-10-12 at 4 50 12 PM" src="https://github.com/hashicorp/design-system/assets/2200899/36e0c4c3-5be5-4cab-a4c4-b61f1f598ff9">

Reply:
yeah I was thinking that too m=but couldn't determine exactly why. (It's not affected by the border.) I just tried shifting by -1px a bit ago but I'll change to -2px.